src/invidious/routes/feeds.cr

```diff
@@ -0,0 +1,434 @@
+
+    templated "feeds/history"
+  end
+
+  # RSS feeds
+
+  def rss_channel(env)
+    locale = LOCALES[env.get("preferences").as(Preferences).locale]?
+
+    env.response.content_type = "application/atom+xml"
+
+    ucid = env.params.url["ucid"]
+
+    params = HTTP::Params.parse(env.params.query["params"]? || "")
+
+    begin
+      channel = get_about_info(ucid, locale)
+    rescue ex : ChannelRedirect
+      return env.redirect env.request.resource.gsub(ucid, ex.channel_id)
+    rescue ex
+      return error_atom(500, ex)
+    end
+
+    response = YT_POOL.client &.get("/feeds/videos.xml?channel_id=#{channel.ucid}")
+    rss = XML.parse_html(response.body)
+
+    videos = rss.xpath_nodes("//feed/entry").map do |entry|
+      video_id = entry.xpath_node("videoid").not_nil!.content
+      title = entry.xpath_node("title").not_nil!.content
+
+      published = Time.parse_rfc3339(entry.xpath_node("published").not_nil!.content)
+      updated = Time.parse_rfc3339(entry.xpath_node("updated").not_nil!.content)
+
+      author = entry.xpath_node("author/name").not_nil!.content
+      ucid = entry.xpath_node("channelid").not_nil!.content
+      description_html = entry.xpath_node("group/description").not_nil!.to_s
+      views = entry.xpath_node("group/community/statistics").not_nil!.["views"].to_i64
+
+      SearchVideo.new({
+        title:              title,
+        id:                 video_id,
+        author:             author,
+        ucid:               ucid,
+        published:          published,
+        views:              views,
+        description_html:   description_html,
+        length_seconds:     0,
+        live_now:           false,
+        paid:               false,
+        premium:            false,
+        premiere_timestamp: nil,
+      })
+    end
+
+    XML.build(indent: "  ", encoding: "UTF-8") do |xml|
+      xml.element("feed", "xmlns:yt": "http://www.youtube.com/xml/schemas/2015",
+        "xmlns:media": "http://search.yahoo.com/mrss/", xmlns: "http://www.w3.org/2005/Atom",
+        "xml:lang": "en-US") do
+        xml.element("link", rel: "self", href: "#{HOST_URL}#{env.request.resource}")
+        xml.element("id") { xml.text "yt:channel:#{channel.ucid}" }
+        xml.element("yt:channelId") { xml.text channel.ucid }
+        xml.element("icon") { xml.text channel.author_thumbnail }
+        xml.element("title") { xml.text channel.author }
+        xml.element("link", rel: "alternate", href: "#{HOST_URL}/channel/#{channel.ucid}")
+
+        xml.element("author") do
+          xml.element("name") { xml.text channel.author }
+          xml.element("uri") { xml.text "#{HOST_URL}/channel/#{channel.ucid}" }
+        end
+
+        videos.each do |video|
+          video.to_xml(channel.auto_generated, params, xml)
+        end
+      end
+    end
+  end
+
+  def rss_private(env)
+    locale = LOCALES[env.get("preferences").as(Preferences).locale]?
+
+    env.response.content_type = "application/atom+xml"
```

Comment:
it could be a good idea to add the following line to Atom feeds handlers:
`env.response.header["Content-Type"] = "application/atom+xml"`

`env.response.content_type` doesn't override the `Content-Type` header, defined in `before_all` as HTML. Because of that, the Atom feeds are not detected as such by the browser.